### PR TITLE
RATEST-302:Add an Identifier for a patient

### PR DESCRIPTION
**Pull request by gracebish**
Jira Issue Number and Title
RATEST-302:Test case: Add an identifier for a patient
Description of changes
Add an identifier for a patient
Issue worked on
https://issues.openmrs.org/browse/RATEST-302

### CODE REVIEW

File: qaframework-bdd-tests/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "postgresCoreInstall": "mvn test -Dcucumber.filter.tags='@selenium and @initialSetup and @postgresInstall'",
     "testingCoreInstall": "mvn test -Dcucumber.filter.tags='@selenium and @initialSetup and @testingInstall'",
     "coreUpgrade": "mvn test -Dcucumber.filter.tags='@selenium and @initialSetup and @upgrade'",
+    "refapp2AddNewIdentifier": "mvn test -Dcucumber.filter.tags='@selenium and @addNewIdentifier'",
```
Comment (Irenyak1) [inline qaframework-bdd-tests/package.json:32]: @gracebish could you move this script to where the refapp2 scripts are.

File: qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AdministrationPage.java [hunk 2]
```diff
@@ -70,6 +71,13 @@ public ManageReportsPage clickOnReportAdministrationLink() {
 		return new ManageReportsPage(this);
 	}
 	
+	public PatientPage clickOnManagePatients() {
+		findElement(MANAGE_PATIENTS_LINK).click();
+		return new PatientPage(this);
+	}
+
+
```
Comment (Irenyak1) [inline qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AdministrationPage.java:79]: @gracebish you only need one empty line here. You could delete the others.

File: qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/PatientFormPage.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	
+	public void setLocation() {
+		selectFrom(LOCATION_FIELD, LOCATION);
+	}
+	
+	public void savePatient() {
+		clickOn(SAVE_PATIENT_BUTTON);
+	}
+}
+
```
Comment (Irenyak1) [inline qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/PatientFormPage.java:50]: @gracebish  try to have just an empty line at the end of the file. These lines are considered as having code/spaces.

**Comment by kdaud** [comment]
@gracebish update your branch with changes from master

**Comment by sherrif10** [comment]
Hello @gracebish  you could resolve conflicting files.

**Comment by gracebish** [comment]
@sherrif10 Not yet. Yesterday i had a sync with @kdaud to help me see whether the tests would run successfully on my windows os and we did not succeed. He slept over it and we are trying to find a solution.

**Comment by sherrif10** [comment]
Sorry about that , Have never tried this on a windows machine before. Is it possible for you to try linux environments it would save you lots of time.

**Review by jwnasambu — CHANGES_REQUESTED** [review]
Kindly ensure `given`  spec is on the same line as other specification in the background section.

```
Background: 
  Given a user clicks on system administration link on the home page
    Then the system loads administration page
    And a user clicks on the manage Patients link on the administration page
    Then the system loads patient page
```

**Review by jwnasambu — COMMENTED** [review]
Kindly ensure `given`  spec is on the same line as other specification in the background section.

```
Background: 
    Given a user clicks on system administration link on the home page
    Then the system loads administration page
    And a user clicks on the manage Patients link on the administration page
    Then the system loads patient page
```

**Comment by gracebish** [comment]
I am still checking out why the IDE presents something different from the commit. Please do not mind this commit.

**Comment by sherrif10** [comment]
Hello @gracebish  Are you still working on this. Any progress.

**Comment by jwnasambu** [comment]
@sherrif10 Her laptop broke down and she is trying to fix it .

**Review by pasindur99 — CHANGES_REQUESTED** [review]
@gracebish Can you add the status badge of this workflow to the QA Dashboard?

**Comment by sherrif10** [comment]
Are you still interested in working on this @gracebish  . You could start with resolving merge conflicts.

**Comment by sherrif10** [comment]
Closing this Pull request.

**Pull request closed without merging**